app/models/manageiq/providers/ibm_cic/inventory/parser/cloud_manager.rb

```diff
@@ -1,2 +1,14 @@
 class ManageIQ::Providers::IbmCic::Inventory::Parser::CloudManager < ManageIQ::Providers::Openstack::Inventory::Parser::CloudManager
+  
+  def parse_vm(vm, hosts)
+    super
+    server = persister.vms.find_or_build(vm.id.to_s)
+    hardware = persister.hardwares.find_or_build(server)
+    # In openstack provider hardware cpu_speed is retrieved from the parent_host.
+    # hardware.cpu_speed = parent_host.try(:hardware).try(:cpu_speed)
+    # In cic provider, cpu_speed is from server details and set to hardware cpu_speed
+    if vm.attributes.fetch("cpu_speed", [])
+      hardware.cpu_speed = vm.attributes.fetch("cpu_speed", [])
+    end
```

Comment:
The way this is now if the cpu_speed is missing we'll always try to set cpu_speed to an empty array which I don't think is what we want, how about :point_down: 
```suggestion
    if vm.attributes.key?("cpu_speed")
      hardware.cpu_speed = vm.attributes["cpu_speed"]
    end
```